src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -141,4 +179,123 @@
         return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
     }
 }
+
+/*
+Function:
+GetLatestJapaneseEraNative
+
+Gets the latest era in the Japanese calendar.
+*/
+int32_t GlobalizationNative_GetLatestJapaneseEraNative(void)
+{
+    @autoreleasepool
+    {
+        // Create an NSCalendar with the Japanese calendar identifier
+        NSCalendar *japaneseCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierJapanese];
+        // Get the latest era
+        NSDateComponents *latestEraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:[NSDate date]];
+        // Extract the era component
+        NSInteger latestEra = [latestEraComponents era];
+        return (int32_t)latestEra;
+    }
+}
+
+/*
+Function:
+GetJapaneseEraStartDateNative
+
+Gets the starting Gregorian date of the specified Japanese Era.
+*/
+int32_t GlobalizationNative_GetJapaneseEraStartDateNative(int32_t era, int32_t* startYear, int32_t* startMonth, int32_t* startDay)
+{
+    @autoreleasepool
+    {
+        NSCalendar *japaneseCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierJapanese];
+        NSDateComponents *startDateComponents = [[NSDateComponents alloc] init];
+        startDateComponents.era = era;
+        // set the date to Jan 1, 1
+        startDateComponents.month = 1;
+        startDateComponents.day = 1;
+        startDateComponents.year = 1;
+        NSDate *date = [japaneseCalendar dateFromComponents:startDateComponents];
+        NSDate *startDate = date;
```

Comment:
Why do we need `startDate`? It seems like they reference the same `NSCalendar` object.

Reply:
It was not needed, removed.